cast/src/lib.rs

```diff
@@ -443,6 +449,77 @@ where
         Ok(self.provider.get_gas_price().await?)
     }
 
+    pub async fn accounts(&self) -> Result<Vec<CastAccount>> {
+        let mut accounts: Vec<CastAccount> = Vec::new();
+
+        // fetch remote accounts
+        if let Ok(remotes) = self.provider.get_accounts().await {
+            for remote in remotes {
+                accounts.push(CastAccount {
+                    address: remote,
+                    kind: CastAccountType::Remote,
+                    path: None,
+                })
+            }
+        }
+
+        // Check the keystore
```

Comment:
This should be able to take an argument similar to `geth --datadir` and fall back to the default keystore if it is not passed. The default keystore is OS dependent